goagen/codegen/workspace.go

```diff
@@ -310,7 +310,7 @@ func PackagePath(path string) (string, error) {
 		if gp, err := filepath.Abs(gopath); err == nil {
 			gopath = gp
 		}
-		if strings.HasPrefix(absPath, gopath) {
+		if filepath.HasPrefix(absPath, gopath) {
```

Comment:
From blame where the note was introduced:

    1. It does not take path boundaries into account.
    2. It assumes that Windows==case-insensitive file system
    and non-Windows==case-sensitive file system, neither of
    which is always true.
    3. Comparing ToLower against ToLower is not a correct
    implementation of a case-insensitive string comparison.
    4. If it returns true on Windows you still don't know how long
    the matching prefix is in bytes, so you can't compute what
    the suffix is.

For `Abs` you have similarly that: "The absolute path name for a given file is not guaranteed to be unique." And, I don't know any Windows API calls that could be used instead either...

From http://www.gamedev.net/topic/628734-cross-platform-filepath-naming-conventions/, 

> Essentially, path comparisons will always be somewhat fragile.

There might be a better way of opening the folders at the same root when the casing doesn't match check whether they are the same file... however I'm not sure whether it will work on networked devices...

Basically, `filepath.HasPrefix` is the devil I know... and I'm not quite sure what the "better way" would be, but it sounds like a very deep rabbit-hole.